docs/ecu-ProtocolServiceBased.js

```
@@ -0,0 +1,414 @@
+    log('error', `Command ${command} failed after ${maxRetries} attempts`);
+
+    return null;
+  }
+
+  // Connection methods
+  async connectToECU() {
+    // Prevent parallel connection attempts
+    if (ProtocolServiceBased.isConnecting) {
+      log('warn', 'Connection attempt already in progress, waiting...');
+      while (ProtocolServiceBased.isConnecting) {
+        await new Promise(resolve => setTimeout(resolve, 100));
```

ESLint

[error] 'setTimeout' is not defined.

(no-undef)